crates/bevy_render/src/texture/image.rs

```diff
@@ -710,6 +1069,15 @@ pub enum TranscodeFormat {
     Rgb8,
 }
 
+/// An error that occurs when accessing specific pixels in a texture
+#[derive(Error, Debug)]
+pub enum TextureAccessError {
+    #[error("out of bounds ({0})")]
+    OutOfBounds(String),
```

Comment:
I see you're using this only with an identical format string, so I'd suggest you add the raw data to the error and format it here:

```suggestion
    #[error("out of bounds (x: {x}, y: {y}, z: {z})")]
    OutOfBounds { x: f32, y: f32, z: f32 },
```